airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/containers.py

```diff
@@ -101,11 +116,16 @@ def format_license_container(dagger_client: dagger.Client, license_file: str) ->
 
 def format_python_container(dagger_client: dagger.Client) -> dagger.Container:
     """Format python code via black and isort."""
-
     return build_container(
         dagger_client,
         base_image=PYTHON_3_10_IMAGE,
         env_vars={"PIPX_BIN_DIR": "/usr/local/bin"},
-        include=["**/*.py", "pyproject.toml", "poetry.lock"],
-        install_commands=["pip install pipx", "pipx ensurepath", "pipx install poetry"],
+        warmup_include=["pyproject.toml", "poetry.lock"],
+        install_commands=[
+            "pip install pipx",
+            "pipx ensurepath",
+            "pipx install poetry",
+            "poetry install --no-root",
```

Comment:
This should speed up the python formatting by caching the `poetry install` which depends on a poetry.lock file which changes very little over time.